Bottom: 07f66efa57022ff8618d8cb4d49d12a268747e51
Top:    3c5ec373323ed7c64b1ad0bfd71cb1181a36a977
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 20:01:09 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 5cbf4ee..356ce0f 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2956,12 +2956,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
